src/utils/emailAndIdExtract.ts

```diff
@@ -6,13 +6,11 @@ type MassMailObject = {
 }
 
 export default function emailAndIdExtract (results: UnprocessedResultsFromCRM[]) {
-    const propertyObjects = results.slice(1)
-
-    const emailsAndIds = propertyObjects.reduce((resultsArray: MassMailObject[], property: UnprocessedResultsFromCRM) => {
-        const contact = property.owner_details.find((owner: OwnerType) => owner.Contact_Type === 'Director')
-        const owner = property.owner_details.find((owner: OwnerType) => owner.Contact_Type === 'Owner')
+    const emailsAndIds = results.reduce((resultsArray: MassMailObject[], property: UnprocessedResultsFromCRM) => {
+        const contact = property.owner_details?.find((owner: OwnerType) => owner.Contact_Type === 'Director')
+        const owner = property.owner_details?.find((owner: OwnerType) => owner.Contact_Type === 'Owner')
         const contactOrOwner = contact || owner
```

Comment:
wouldn't you want to include all owners and contacts rather than only one of them? I think this should be something like 

```
const emailsAndIds = results.flatMap((property: UnprocessedResultsFromCRM) => {
   return property.owner_details?.map((ownerOrContact) => {
     return {
       email: ownerOrContact.Email,
        id: ownerOrContact.id
     };
   }) || []
})
```